frontend/src/components/NewRunParametersV2.tsx

```diff
@@ -91,13 +107,45 @@ function convertInput(paramStr: string, paramType: ParameterType_ParameterTypeEn
   }
 }
 
+function checkInput(
+  parametersInRealType: any,
+  paramType: ParameterType_ParameterTypeEnum,
+): { validInput: boolean; errorMessage: string } {
+  let isValid: boolean;
+  let eMsg: string;
+  switch (parametersInRealType) {
+    case undefined:
+      isValid = false;
+      eMsg = 'Missing parameter.';
+      break;
+    // TODO(jlyaoyuli): tell the error difference between mismatch type or invalid JSON form.
+    case null:
+      isValid = false;
+      eMsg =
+        'Invalid input. This parameter should be in ' +
+        ParameterType_ParameterTypeEnum[paramType] +
+        ' type';
+      break;
+    default:
+      isValid = true;
+      eMsg = '';
+  }
+  return {
+    validInput: isValid,
+    errorMessage: eMsg,
+  };
+}
+
 function NewRunParametersV2(props: NewRunParametersProps) {
   const [customPipelineRootChecked, setCustomPipelineRootChecked] = useState(false);
   const [customPipelineRoot, setCustomPipelineRoot] = useState(props.pipelineRoot);
+  const [validInputs, setValidInputs] = useState([]);
+  const [errorMessages, setErrorMessages] = useState(['']);
```

Comment:
This `array of string` is used to store error message. The message is either `invalid input` or `missing parameter`; otherwise, it will be empty string for correct input type.

Reply:
Let's say there is a pipeline with no parameter, then we don't need to show any error message or empty message, right? My question is more about why are we assuming there is one parameter by setting `useState([''])`. We should assume the pipeline has no parameter, thus we set `useState([])`.

Reply:
Great example. Thank you James.